### DISCUSSION TITLE
Navigating between pages

### DISCUSSION THREAD

**Discussion by KadeemCampbell** [Budibase/budibase#4124] [Help] [answered]
Good day. I would love some assistance with an issue. 

Let's say I have a table of data about soccer teams. I have designed two screens. One screen allows me to add another row to the table and the other screen allows me to view the info for the row that I just created. 

On the Add Team page I have a button to save the row and navigate to the View Team page. How do I define what path to go to when defining the actions so that I see the record I just created on the View Team page?

**Comment by yatharth-zeotap** [comment]
Even I am trying to do something similar, taking a table row element and trying to navigate to another page with that element auto-populating some field. 
Any help would be great !!

**Comment by mjashanks** [comment, marked as answer]
Hi @yatharth-zeotap @KadeemCampbell 

You can now use the results from previous button action steps. I.e you can `Navigate to` the saved row...

![image](https://user-images.githubusercontent.com/3524181/150540456-3e55e3fc-ddb6-4d44-b0d1-4fa013fd3f47.png)

In my example my steps are
1.. `Save Row`
2.  `Navigate To` - `/contact/{{ Action 1.Saved Row._id }}`

**Reply by KadeemCampbell** [reply]
Thank you very much @mjashanks . This definitely solved my problem. I kept seeing the option in the bindable values but never tried it.

**Comment by yatharth-zeotap** [comment]
Hi @mjashanks 

My use case is a bit different.
Consider I have a screen like this.
![image](https://user-images.githubusercontent.com/74049974/150635344-fac40735-2c96-4f83-af97-31e5743d5c59.png)
And the 2nd screen to be this

![image](https://user-images.githubusercontent.com/74049974/150635354-1b47980e-936b-454a-922f-00755ccc0ce1.png)

Now on using the button I want to open a modal and populate the value from `COLUMN1` to the text field and `COLUMN2` to the paragraph field.
So in this case what should be the URL that I should provide in the `Navigate To` action.

This is my UI structure.
![image](https://user-images.githubusercontent.com/74049974/150635374-d3dcbf07-af56-46a3-9aed-9980c5e8a385.png) ![image](https://user-images.githubusercontent.com/74049974/150635379-6b2881ac-425b-4009-ab8d-0183acbeb814.png)

**Reply by mjashanks** [reply]
@yatharth-zeotap so you would like to display an "Edit" screen, in a modal?

So, the screen that is to appear in the modal should have a URL like `/your-record/:id` the `:id` is a URL parameter, which we will use to load the record.

Your edit screen should have a structure a bit like this:

![image](https://user-images.githubusercontent.com/3524181/150768329-b79f8e82-14e7-4f22-ae74-df11658a56bb.png)

I.e. 
```
> Data Provider
  > Repeater 
     > Form
            ....fields
```

Your data provider should load the record from your table, and should have a filter like this:

![image](https://user-images.githubusercontent.com/3524181/150768505-4dd5f0a6-d5bf-42ad-96eb-124c241da207.png)

I.e. it's loading the record based on the `:id` passed in via the URL parameter.

That should be enough to get you going! Your Save button will need to include a `Close Screen Modal` action, to close the modal.

**Reply by enpro-github** [reply]
@mjashanks I have similar issue but your suggestions didn't work for me.  My datasource is a RestAPI where there is a field already called "id". 
 I have no field called "_id" anywhere that i can see. Is there another way to do this?

**Reply by mjashanks** [reply]
`_id` is a field that is specific to Budibase's internal DB.

You can use whatever field you like in here - i.e. any field that you can use to fetch your record. 

`id` will work... i.e. presumably you have a REST enpoint that allows you to fetch a single record by ID, e.g. `GET /my-records/4` (where `4` is the `id`)

**Reply by enpro-github** [reply]
@mjashanks i've followed your instructions multiple times, and attempted work arounds using a button and the state,  but i am unable to get this to work.  Every instance results in a blank page due to no match, or I get redirected to  the  /:id url, not the url specific to the record..

**Reply by mjashanks** [reply]
Hey @enpro-github. Attached below is an example app, which you can import. I have
1. A REST query "List Repos" which lists GitHub repositories for Budibase
2. A REST query that fetches a single repository
3. ON my home screen, I use the "List Repos" to display all repositories in a table.
4. The table has a button, which navigates to: `/repos/{{ New Table.List Repos.name }}`, in  a modal
5. I have a second screen `/repos/:name` (notice I'm using a `name` parameter - it can be called whatever you like)
6. In this screen, I have a data provider, which passes the `{{ URL.name }}` value into my rest query (using the gear icon).

[GitHub Repos-export-1643883561067.txt](https://github.com/Budibase/budibase/files/7993647/GitHub.Repos-export-1643883561067.txt)

**Reply by enpro-github** [reply]
@mjashanks Thank you for that.  I followed your example, and it works like a charm
